packages/api/cms-api/src/user-permissions/user-permissions.types.ts

```diff
@@ -2,9 +2,9 @@ import { ModuleMetadata, Type } from "@nestjs/common";
 
 import { CurrentUser } from "./dto/current-user";
 import { FindUsersArgs } from "./dto/paginated-user-list";
-import { User } from "./dto/user";
```

Comment:
I wouldn't remove the DTO. Let's suppose I want to be able to query the additional fields in the application: Then I'd need to add a field resolver for the user object that returns the fields. For this I would need this DTO.

Reply:
> Wouldn't it be sufficient exporting the DTO so that it can be used in the annotation of the field resolver?

Yes, that's what I've meant. Should have been "I wouldn't remove the _export of the_ DTO".

> We could do that but under what name? User should be the interface so that it can easily be augmented.

Isn't it possible to augment a class as well? Also, I believe it's possible to have both a class and an interface with the same name as exports.

Reply:
> Isn't it possible to augment a class as well?

of course not, it's crazy enough for interfaces :)

Reply:
Is changing `User` from a class to an interface a breaking change?

Reply:
> Said this, I doubt if 50b9bc0 even makes sense.

Then let's revert it and only add it to the public API once we need it.